lib/os/timeutil.c

```diff
@@ -36,8 +36,8 @@ static int64_t time_days_from_civil(int64_t y,
 	}
 
 	int64_t era = (y >= 0 ? y : y - 399) / 400;
-	unsigned int yoe = y - era * 400;
-	unsigned int doy = (153U * (m + (m > 2 ? -3 : 9)) + 2U) / 5U + d;
+	unsigned int yoe = (unsigned int)(int64_t)(y - era * 400);
+	unsigned int doy = (153U * (m > 2 ? m - 3 : m + 9) + 2U) / 5U + d;
```

Comment:
`m > 2U`, `m - 3U`, `m + 9U`?

Reply:
It has been asked to reduce the number of introduced U suffix to the bare minimum and I have seconded that.